src/lib/io/sol-iio.c

```diff
@@ -157,6 +157,8 @@ craft_filename_path(char *path, size_t size, const char *base, ...)
     len = vsnprintf(path, size, base, args);
     va_end(args);
 
+    SOL_DBG("Crafted path len: %d Buffer size: %zu", len, size);
```

Comment:
the important information should be the path, which is not printed. You can shrink it to: "available=%zu, used=%d, path=\"%s\""

You can omit "crafted" since it's going to show the function name.